AWSGameLift.Server Module Causes AZ_TRAIT_CLIENT = TRUE

Adding AWSGameLift.Server.Static as a build dependency causes AZ_TRAIT_CLIENT to be set to true, resulting in compiler errors.

**Steps to reproduce**
1. Create a new game using MultiplayerTemplate
2. Enable the GameLift Gem
3. Open MyGame\Gem\CMakeLists.txt
4. Add Gem::AWSGameLift.Server.Static to ${gem_name}.Server.Static STATIC > BUILD_DEPENDENCIES > PRIVATE 
![image](https://github.com/user-attachments/assets/c8273e01-7e79-41ff-8d21-f7c66227fb75)
5. Compile MyGame.Server.Static 
6. See compiler error

**Expected behavior**
Server module compiles

**Actual behavior**
Compiler error
```
>\CoolGameLiftGame\Gem\Source\Components\NetworkSimplePlayerCameraComponent.cpp(18,1): 
     error C1083: Cannot open include file: 'DebugDraw/DebugDrawBus.h': No such file or directory
```

**Found in Branch**
Development / 24.09 release

commit fa80fab682ad429596e0daf4aa41e3846396e563
Sat Nov 2 13:38

--- a/Gems/AWSGameLift/Code/AWSGameLiftServer/CMakeLists.txt
+++ b/Gems/AWSGameLift/Code/AWSGameLiftServer/CMakeLists.txt
@@ -36,7 +36,7 @@ ly_add_target(
     BUILD_DEPENDENCIES
         PRIVATE
             AZ::AzCore
-            Gem::Multiplayer.Unified.Static
+            Gem::Multiplayer.Server.Static
             3rdParty::AWSGameLiftServerSDK
     )
 
@@ -83,7 +83,7 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
             PRIVATE
                 AZ::AzCore
                 AZ::AzTest
-                Gem::Multiplayer.Unified.Static
+                Gem::Multiplayer.Server.Static
                 Gem::${gem_name}.Server.Static
                 3rdParty::AWSGameLiftServerSDK
     )

